rqt_moveit/src/rqt_moveit/moveit_widget.py

```diff
@@ -150,34 +139,69 @@ def _check_nodes_alive(self, signal, nodes_monitored):
 
         @param signal: Signal(bool, str)
         @type nodes_monitored: str[]
+        @type stop_event: Event()
         """
-        while self._is_checking_nodes:
-            rosnode_dynamically_loaded = __import__('rosnode')
-            #from rosnode import rosnode_ping
+        rosnode_dynamically_loaded = __import__('rosnode')
+        while True:
             for nodename in nodes_monitored:
-                #TODO: rosnode_ping prints when the node is not found.
-                # Currently I have no idea how to capture that from here.
+                is_node_running = False
                 try:
-                    #is_node_running = rosnode_ping(nodename, 1)
-                    is_node_running = rosnode_dynamically_loaded.rosnode_ping(
-                                                                 nodename, 1)
+                    registered_nodes = rosnode_dynamically_loaded.get_node_names()
+                    if nodename in registered_nodes:
+                        is_node_running = True
```

Comment:
These two lines could be written as `is_node_running = nodename in registered_nodes` which would also make line 147 obsolete.